internal/service/datazone/project.go

```diff
@@ -0,0 +1,452 @@
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			names.AttrDescription: schema.StringAttribute{
+				Optional: true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtMost(2048),
+				},
+			},
+			"domain_id": schema.StringAttribute{
+				Required: true,
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexache.MustCompile(`^dzd[-_][a-zA-Z0-9_-]{1,36}$`), "must conform to: ^dzd[-_][a-zA-Z0-9_-]{1,36}$ "),
+				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+
+			"glossary_terms": schema.ListAttribute{
+				CustomType:  fwtypes.ListOfStringType,
+				ElementType: types.StringType,
+
+				Validators: []validator.List{
+					listvalidator.SizeBetween(1, 20),
+					listvalidator.ValueStringsAre(stringvalidator.RegexMatches(regexache.MustCompile(`^[a-zA-Z0-9_-]{1,36}$`), "must conform to: ^[a-zA-Z0-9_-]{1,36}$ ")),
+				},
+				Optional: true,
+			},
+
+			names.AttrName: schema.StringAttribute{
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexache.MustCompile(`^[\w -]+$`), "must conform to: ^[\\w -]+$ "),
+					stringvalidator.LengthBetween(1, 64),
+				},
+				Required: true,
+			},
+			"created_by": schema.StringAttribute{
+				Computed: true,
+			},
+			names.AttrID: framework.IDAttribute(),
+
+			names.AttrCreatedAt: schema.StringAttribute{
+				CustomType: timetypes.RFC3339Type{},
+				Computed:   true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+
+			"failure_reasons": schema.ListAttribute{
+				CustomType: fwtypes.NewListNestedObjectTypeOf[dsProjectDeletionError](ctx),
+				Computed:   true,
+			},
+
+			"last_updated_at": schema.StringAttribute{
+				CustomType: timetypes.RFC3339Type{},
+				Computed:   true,
+			},
+			"project_status": schema.StringAttribute{
+				CustomType: fwtypes.StringEnumType[awstypes.ProjectStatus](),
+				Computed:   true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"skip_deletion_check": schema.BoolAttribute{
+				Optional: true,
+				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+		Blocks: map[string]schema.Block{
+			names.AttrTimeouts: timeouts.Block(ctx, timeouts.Opts{
+				Create: true,
+				Delete: true,
+			}),
+		},
+	}
+}
+
+func (r *resourceProject) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan resourceProjectData
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	conn := r.Meta().DataZoneClient(ctx)
+
+	var validateDomain datazone.GetDomainInput
+	validateDomain.Identifier = plan.DomainId.ValueStringPointer()
+	_, err := conn.GetDomain(ctx, &validateDomain)
```

Comment:
we should avoid doing read operations in the `Create` handler since the single operation here should be to create. In this case we can count on the AWS API returning an appropriate error if the `domain_id` is not correct when creating.